### PR TITLE
[nginx] special handling for rate-limited traffic on the catalog

### LINKED ISSUE #4073: Add responsible rate limiting to the catalog
We would like to block badly-behaved IPs that hit our main catalog (generally bots, see #4071 and #4069). However, we want to enable users to find items and data in our systems. To make catalog rate limiting effective but also humane:

- [x] limit requests to 5 per second - this is as fast as any human can click - related to #4072 
- [x] allow internal IPs unlimited access to enable integrations between our sites
- [x] research how to redirect requests that hit our rate limits to a page that describes alternate ways to download large data sets - related to https://github.com/pulibrary/orangelight/issues/3638

### PR DESCRIPTION
**Pull request by acozine**
Closes #4073.

Needs to be tested. Also, we should probably merge #4121 first, then rebase this PR.

- Configures nginx to return `429 Too Many Requests` instead of the default `503 Service Unavailable` when a client hits the `catalog-prod-ratelimit`.
- Adds a new custom error page for `429` errors that links to the Open Dataset page.
- Directs `429` errors to the new custom error page.

### CODE REVIEW

**Comment by acozine** [comment]
Partially addresses https://github.com/pulibrary/orangelight/issues/3638

**Comment by acozine** [comment]
Tested successfully today. To test rate-limiting, we:
1. installed `siege` on a laptop with `brew install siege`
2. disconnected from the VPN (VPN traffic is never rate-limited)
3. created a file called `combined` that contained a long list of URLs in the catalog
4. ran `siege --delay=0.1 --file=combined --internet --verbose --reps=100 --concurrent=20`
5. while the `siege` command was still running, opened a catalog page in a browser to view the error results

File: roles/nginxplus/files/ratelimit.html [hunk 1]
```diff
@@ -73,18 +73,14 @@
       <div class="wrapper">
         <h1>You have sent too many requests!</h1>
         <p>
-          We have rate-limited your traffic temporarily. If you are trying to download a large amount of data, try our Open Dataset. If you think your traffic should be allowed, contact Library Support.
+          We have rate-limited your traffic temporarily. If you are trying to download a large amount of data, <a href="https://catalog.princeton.edu/dataset">try our Open Dataset</a>. If you think your traffic should be allowed, contact Library Support.
```
Comment (christinach) [inline roles/nginxplus/files/ratelimit.html:76]: `try our <a href="https://catalog.princeton.edu/dataset">Open Dataset</a>`

**Review by christinach — COMMENTED** [review]
@acozine I added one comment to move 'try our' out of the href text.

**Comment by christinach** [comment]
works great thank you!

File: roles/nginxplus/files/ratelimit.html (hunk 1 above)
Comment (acozine) [reply roles/nginxplus/files/ratelimit.html]: Thanks! Addressed in 1c0fd539ba2f6e1bdf791b3a8d1ced2953d91474.

**Pull request merged**